### PR TITLE
Polish QR-Code animation

### PR DESCRIPTION
**Pull request by Schedulaar**
Whoopsiedaisy

Let me know whether you like, and what to improve :D

https://user-images.githubusercontent.com/7025199/110248854-5b9e4e00-7f73-11eb-91ec-a0c38400881a.mp4

### CODE REVIEW

**Review by elKei24 — APPROVED** [review]
WOW

File: frontend/lib/identification/rectangular_progress_indicator_painter.dart [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+        path.relativeLineTo(min(remainingPath, horizontalLineLength), 0);
+        remainingPath = max(remainingPath - horizontalLineLength, 0);
+      }
+      {
+        // Bottom right arc
+        var angle = min(remainingPath / quarterArcLength, 1.0) * pi / 2;
+        var dest =
+            Offset(strokeRadius * sin(angle), -strokeRadius * (1 - cos(angle)));
+        path.relativeArcToPoint(dest,
+            radius: Radius.circular(strokeRadius), clockwise: false);
```
Comment (RoAnBu) [inline frontend/lib/identification/rectangular_progress_indicator_painter.dart:79]:
```suggestion
        path.relativeArcToPoint(dest,
            radius: Radius.circular(strokeRadius), clockwise: false);
        remainingPath = max(remainingPath - quarterArcLength, 0);
```
I think you forgot something here

Comment (RoAnBu) [reply frontend/lib/identification/rectangular_progress_indicator_painter.dart:79]: image

Causing this critical bug 😱🤯

**Review by RoAnBu — APPROVED** [review]
Unfortunately, I have no fancy gif here 😢 .

But damn, it's smoooooth!

**Review by maxammann — APPROVED** [review]

**Pull request merged**